crm_rma_location/models/stock_warehouse.py

```diff
@@ -0,0 +1,161 @@
+            customer_loc, supplier_loc = self._get_partner_locations()
+
+            color = self.compute_next_color()
+
+            # order the picking types with a sequence
+            # allowing to have the following suit for
+            # each warehouse: reception, internal, pick, pack, ship.
+            max_sequence = picking_type_model.search_read(
+                [], ['sequence'], order='sequence desc'
+            )
+            max_sequence = max_sequence and max_sequence[0]['sequence'] or 0
+
+            in_type_id = warehouse.rma_in_type_id
+            if not in_type_id:
+                in_type_id = picking_type_model.create(vals={
+                    'name': _('RMA Receipts'),
+                    'warehouse_id': warehouse.id,
+                    'code': 'incoming',
+                    'sequence_id': in_seq_id.id,
+                    'default_location_src_id': customer_loc.id,
+                    'default_location_dest_id': wh_stock_loc.id,
+                    'sequence': max_sequence + 4,
+                    'color': color})
+
+            out_type_id = warehouse.rma_out_type_id
+            if not out_type_id:
+                out_type_id = picking_type_model.create(vals={
+                    'name': _('RMA Delivery Orders'),
+                    'warehouse_id': warehouse.id,
+                    'code': 'outgoing',
+                    'sequence_id': out_seq_id.id,
+                    'return_picking_type_id': in_type_id.id,
+                    'default_location_src_id': wh_stock_loc.id,
+                    'default_location_dest_id': supplier_loc.id,
+                    'sequence': max_sequence + 1,
+                    'color': color})
+                in_type_id.write({'return_picking_type_id': out_type_id.id})
+
+            int_type_id = warehouse.rma_int_type_id
+            if not int_type_id:
+                int_type_id = picking_type_model.create(vals={
+                    'name': _('RMA Internal Transfers'),
+                    'warehouse_id': warehouse.id,
+                    'code': 'internal',
+                    'sequence_id': int_seq_id.id,
+                    'default_location_src_id': wh_stock_loc.id,
+                    'default_location_dest_id': wh_stock_loc.id,
+                    'active': True,
+                    'sequence': max_sequence + 2,
+                    'color': color})
+
+            # write picking types on WH
+            warehouse.write({
+                'rma_in_type_id': in_type_id.id,
+                'rma_out_type_id': out_type_id.id,
+                'rma_int_type_id': int_type_id.id,
+            })
+
+    @api.multi
+    def create_locations_rma(self):
+        """
+        Create a RMA location for RMA movements that takes place when internal,
+        outgoing or incoming pickings are made from/to this location
+        """
+        location_obj = self.env['stock.location']
+        context_with_inactive = self.env.context.copy()
```

Comment:
No need to copy the context. When using `with_context` the kwargs are overrides.

From `openerp/models.py`

```python
            # current context is {'key1': True}
            r2 = records.with_context({}, key2=True)
            # -> r2._context is {'key2': True}
            r2 = records.with_context(key2=True)
            # -> r2._context is {'key1': True, 'key2': True}
```